### PR TITLE
[Reviewer: Alex] Fix up monit install issues

### PR DESCRIPTION
**Pull request by eleanor-merry**
This PR fixes up some issues with the monit install (e.g. setting the config file early, breaking a dependency on clearwater-infrastructure,

### CODE REVIEW

File: debian/clearwater-monit.service [hunk 1]
```diff
@@ -4,7 +4,7 @@
 Description=Clearwater-monit service manager
 
 [Service]
-ExecStartPre=/usr/share/clearwater/bin/issue-alarm "systemd" "4500.6"
+ExecStartPre=[ -x /usr/share/clearwater/bin/issue-alarm ] && /usr/share/clearwater/bin/issue-alarm "systemd" "4500.6" || true
```
Comment (AlexHockey) [inline debian/clearwater-monit.service]: Why would this script ever not be executable. Isn't that a bug in itself?

File: rpm/clearwater-monit.spec [hunk 2]
```diff
@@ -24,6 +24,7 @@ build_files_list > clearwater-monit.files
 if [ "$1" == 1 ] ; then
   /usr/share/clearwater/clearwater-monit/install/clearwater-monit.postinst
 fi
+chmod 600 /etc/monit/monitrc
```
Comment (AlexHockey) [inline rpm/clearwater-monit.spec]: How come we need to chmod on centos but not on debian?

File: debian/clearwater-monit.service (hunk 1 above)
Comment (eleanor-merry) [inline debian/clearwater-monit.service]: Yes - but in a different package (so it shouldn't stop the clearwater-monit install)

File: rpm/clearwater-monit.spec (hunk 2 above)
Comment (eleanor-merry) [inline rpm/clearwater-monit.spec]: We do - it's done at https://github.com/Metaswitch/clearwater-monit/blob/master/debian/rules#L19

**Comment by AlexHockey** [comment]
Looks good. A couple of comments inline. How have you tested this PR?

File: debian/clearwater-monit.service (hunk 1 above)
Comment (AlexHockey) [inline debian/clearwater-monit.service]: We discussed. `issue-alarm` is installed by clearwater-infrastructure which depends on clearwater monit for some reason. This feels like we've got our dependencies wrong - I would expect clearwater infrastructure to be at the "bottom" of the dependency graph, and I can't immediately think of anything in clearwater infrastructure that uses monit. You are going to check the code in as-is and investigate the dependency issue separately.

**Pull request merged**